backend/src/bundles/users/user.entity.ts

```diff
@@ -9,78 +9,93 @@ class UserEntity implements Entity {
 
     private 'passwordSalt': string;
 
+    private 'fullName': string | null;
+
     private constructor({
         id,
         email,
         passwordHash,
         passwordSalt,
+        fullName,
     }: {
         id: number | null;
         email: string;
         passwordHash: string;
         passwordSalt: string;
+        fullName: string | null;
     }) {
         this.id = id;
         this.email = email;
         this.passwordHash = passwordHash;
         this.passwordSalt = passwordSalt;
+        this.fullName = fullName;
     }
 
     public static initialize({
         id,
         email,
         passwordHash,
         passwordSalt,
+        fullName,
     }: {
         id: number;
         email: string;
         passwordHash: string;
         passwordSalt: string;
+        fullName: string | null;
     }): UserEntity {
         return new UserEntity({
             id,
             email,
             passwordHash,
             passwordSalt,
+            fullName,
         });
     }
 
     public static initializeNew({
         email,
         passwordHash,
         passwordSalt,
+        fullName,
     }: {
         email: string;
         passwordHash: string;
         passwordSalt: string;
+        fullName: string | null;
     }): UserEntity {
         return new UserEntity({
             id: null,
             email,
             passwordHash,
             passwordSalt,
+            fullName,
```

Comment:
looks like we dont need it here

Reply:
Could you explain why, please? Because I thought we would operate with a user with all the possible fields, so that the user was shaped something like this
```
user: {
  id: 1,
  email: 'email,
  fullName: 'Full Name',
}
```
and not like this
```
user: {
  id: 1,
  email: 'email,
  userDetails: {
    fullName: 'Full Name',
  }
}
```

Reply:
- initializeNew (create operation): This method initializes a new entity with only the necessary fields for creating an entity in the database.
- toNewObject (paired with initializeNew): This method converts the initialized entity to object for creating entity in the database
---
- initialize (creating the whole entity): This method initializes a new entity with all fields for the entity - should be returned from repository
- toObject (creating DTO from entity): This method converts an entity into a DTO

So when using initializeNew for entity creation, you don't need to pass fullName, and toNewObject should not return fullName in the object it creates.

Reply:
Ohhh, I get it now. Thanks for the clarification